src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -1092,7 +1090,8 @@ bool emberAfOperationalCredentialsClusterCSRRequestCallback(app::CommandHandler
 
         err = Credentials::ConstructNOCSRElements(ByteSpan{ csrSpan.data(), csrSpan.size() }, CSRNonce, kNoVendorReserved,
                                                   kNoVendorReserved, kNoVendorReserved, nocsrElementsSpan);
-        VerifyOrExit((err == CHIP_NO_ERROR) && (nocsrElementsSpan.size() <= kMaxRspLen), finalStatus = Status::Failure);
+        VerifyOrExit((err == CHIP_NO_ERROR) && (nocsrElementsSpan.size() <= Credentials::kMaxRspLen),
```

Comment:
This is part of the contract of ConstructNOCSRElements: it is not supposed to make the span larger than kMaxRspLen. The check should be in ConstructNOCSRElements

Reply:
moved this check into DeviceAttestationConstructor.cpp